File: core/MoveZeroes/MoveZeroes.java
package com.dbc;

public class MoveZeroes {
    public void moveZeroes(int[] nums) {
        int point = 0, cur = 0;
        while (cur < nums.length) {
            while (cur < nums.length && nums[cur] == 0) cur++;
            if (cur < nums.length) {
                int temp = nums[point];
                nums[point] = nums[cur];
                nums[cur] = temp;
                cur++;
                point++;
            }
        }
    }
}

File: core/MoveZeroes/MoveZeroes.py
from typing import List


def moveZeroes(nums: List[int]) -> None:
    point, cur, length = 0, 0, len(nums)
    while cur < length:
        while cur < length and nums[cur] == 0:
            cur += 1
        if cur < length:
            nums[point], nums[cur] = nums[cur], nums[point]
            point += 1
            cur += 1
